### ISSUE TITLE
Tmux Copy/Paste/Scroll in Darwin

### ISSUE THREAD

**Issue by expelledboy** [NixOS/nixpkgs#71810]
**Describe the bug**

I am failing to get copy-paste and scrolling mouse behaviors working under nix. Granted I am very fresh to nix, and attempting an install on my mac, but I have honestly been hacking away at this for 2 days.

**To Reproduce**

I have Alactritty installed using `brew cask install alactritty`.

Which I configure using `home-manager` to use tmux.

```nix
  xdg.configFile = {
    "alacritty/alacritty.yml".text = ''
      shell:
        # Use nix installed tmux.
        program: ${pkgs.tmux}/bin/tmux

      ${builtins.readFile ./alacritty.yml}
    '';
  };
```

I configure tmux with the `tmux-yank` plugin.

```nix
  programs.tmux = {
    enable = true;

    # Use vim keybinding in copy mode.
    keyMode = "vi";

    plugins = with pkgs.tmuxPlugins; [
      yank
    ];

    extraConfig = ''
      # Use zsh as the default shell without installing it in the user.
      set-option -g default-shell "${pkgs.zsh}/bin/zsh"
      set-option -g default-command "${pkgs.reattach-to-user-namespace}/bin/reattach-to-user-namespace -l ${pkgs.zsh}/bin/zsh"

      ${builtins.readFile ./tmux.conf}
    '';
  };
```

This is what can be found in `tmux.conf`

```tmux
# Mouse mode on
set -g terminal-overrides 'xterm*:smcup@:rmcup@'
set -g mouse on
```

Install `neovim` with appropriate plugins for tmux integration.

```nix
  # Configure vim.
  programs.neovim = {
    plugins = with pkgs.vimPlugins; [
      # Tmux integration.
      vim-tmux-focus-events
      vim-tmux-navigator
      vim-tmux-clipboard
    ];
  }
```

**Expected behavior**

- Mouse scroll up in at zsh prompt to enter tmux copy mode.
  Highlight text to copy selection into system clipboard.
- Mouse scroll up in command like `man tree` to scroll buffer.
- Mouse scroll up in vim to move vim cursor.
- Highlight text while in vim to copy selection into system clipboard.

**Metadata**

As a side note, I am getting the following error when running the suggested command for nix-info:

```sh
nix run nixpkgs.nix-info -c nix-info -m
Assertion failed: (size_ < capacity_), function push_back, file src/libexpr/attr-set.hh, line 54.
```

But jumping into a nix-shell -p nix-info, here is the meta:

 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.1`
 - channels(anthony): `"home-manager, nixpkgs-20.03pre196811.7818f30cc4b"`
 - nixpkgs: `/Users/anthony/.nix-defexpr/channels/nixpkgs`

**Comment by eraserhd** [comment]
Have you installed `reattach-to-user-namespace`?  Clipboard access does not work in shells in tmux under Mac OS without it.  It looks like the `yank` plugin will detect it in the path automatically, but I would recommend setting `reattach-to-user-namespace -l my-shell` as tmux's `default-command`.

If that's not it, what happens instead of the "Expected behavior"?

**Comment by expelledboy** [comment]
No copying occurs. I have tried setting the `default-command` to use `reattach-to-user-namespace`, both the one I installed using brew and nix.

Does anyone have a working example with nix+darwin+tmux(+alactritty)?

**Comment by eraserhd** [comment]
I just now opened tmux in Alacritty (installed from Nix a few weeks ago), manually ran `set -g mouse on` and was able to scroll my shell history with the trackpad in bash and select text.  I have no `alacritty.yml` at the moment, since I just switched from using it normally.  I have tmux next-3.1 from [this override](https://github.com/eraserhd/dotfiles/blob/develop/dogfood/default.nix#L38-L41).

When I was using Alacritty day-to-day (up to about a week ago), I occasionally needed to copy from the terminal.  If the application in the tmux window "wanted" the mouse, Alacritty wouldn't let me select text with the trackpad.  This is Alacritty's copy mode, not tmux's.  (I use tmux copy mode from the keyboard normally.)

I don't have yank installed.  My tmux config is [here](https://github.com/eraserhd/dotfiles/blob/develop/tmux/default.nix).

My $TERM in Alacritty (the one tmux sees) is `xterm-256color`.
My $TERM inside tmux is `tmux-256color`, but is patched and manually installed for palette colors from the `install.sh` and tic files [here](https://github.com/eraserhd/dotfiles/tree/develop/tmux), though I _think_ this script is broken because the two versions of ncurses on my system compete in weird ways that I might have sorted out manually.

I hope this helps.
